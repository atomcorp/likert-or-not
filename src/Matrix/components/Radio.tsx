import * as React from 'react';
import css from '../Matrix.css';
import {Radio} from '../definitions';

const Radio = (props: Radio) => (
  <input
    className={css.radio}
    name={props.name}
    value={props.value}
    type="radio"
    checked={props.checked}
    onChange={(e) =>
      props.handleClick(e, {
        id: props.name,
        value: props.value,
      })
    }
  />
);

export default Radio;
