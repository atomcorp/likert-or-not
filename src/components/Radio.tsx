import * as React from 'react';
import css from '../index.css';
import LikertOrNot from '../index.d';

const Radio = (props: LikertOrNot.Radio) => (
  <label className={css.radio}>
    <input
      className={css.input}
      name={props.name}
      value={props.value}
      type="radio"
      checked={props.checked}
      onChange={(e: React.SyntheticEvent) =>
        props.handleClick(e, {
          id: props.name,
          value: props.value,
        })
      }
    />
    <span className={css.inlineLabel}>{props.header}</span>
  </label>
);

export default Radio;
