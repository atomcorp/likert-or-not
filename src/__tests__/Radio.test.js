import React from 'react';
import Radio from '../components/Radio';
import renderer from 'react-test-renderer';

test('Title prints text in caption', () => {
  const mockClick = jest.fn((id, value) => ({
    id,
    value
  }))
  const component = renderer.create(
    <Radio 
      name="Radio name"
      value="A"
      checked={true}
      handleClick={() => mockClick('Radio name', 'A')}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
  // click the <Radio />
  tree.props.onChange();
  expect(mockClick.mock.calls[0]).toEqual([ "Radio name", "A" ])
  // re-render
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
