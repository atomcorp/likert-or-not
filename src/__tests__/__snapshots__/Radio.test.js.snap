// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title prints text in caption 1`] = `
<input
  checked={true}
  className={undefined}
  name="Radio name"
  onChange={[Function]}
  type="radio"
  value="A"
/>
`;

exports[`Title prints text in caption 2`] = `
<input
  checked={true}
  className={undefined}
  name="Radio name"
  onChange={[Function]}
  type="radio"
  value="A"
/>
`;
